src/main/java/org/minions/devfund/marylin/WhackAMole.java

```diff
@@ -0,0 +1,158 @@
+
+    /**
+     * Constructor of WhackAMole class.
+     *
+     * @param numAttempts   Number of the attempts.
+     * @param gridDimension Number of the grid dimension.
+     */
+    WhackAMole(int numAttempts, int gridDimension) {
+        this.moleGrid = new char[gridDimension][gridDimension];
+        this.attemptsLeft = numAttempts;
+
+        for (char[] row : this.moleGrid) {
+            Arrays.fill(row, '*');
+        }
+    }
+
+    /**
+     * Method to place a mole into the grid.
+     *
+     * @param x X axis in the grid.
+     * @param y Y axis in the grid.
+     * @return Returns true if the mole was place, false otherwise.
+     */
+    boolean place(int x, int y) {
+        if (this.moleGrid[x][y] != 'M') {
+            this.moleGrid[x][y] = 'M';
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to whack a mole in the grid.
+     *
+     * @param x X axis in the grid.
+     * @param y Y axis in the grid.
+     */
+    void whack(int x, int y) {
+        if (this.moleGrid[x][y] == 'M') {
+            this.moleGrid[x][y] = 'W';
+            this.score++;
+            this.molesLeft--;
+        }
+        this.attemptsLeft--;
+    }
+
+    /**
+     * Method to print the status of the game without shows the moles.
+     */
+    void printGridToUser() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                if (this.moleGrid[i][j] == 'M') {
+                    System.out.print("*  ");
+                } else {
+                    System.out.print(this.moleGrid[i][j] + "  ");
+                }
+            }
+            System.out.println("");
+        }
+    }
+
+    /**
+     * Method to print the grid showing the status of the game even moles.
+     */
+    void printGrid() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                System.out.print(this.moleGrid[i][j] + "  ");
+            }
+            System.out.println("");
+        }
+    }
+
+    /**
+     * Main method to play the game.
+     *
+     * @param args Default Argument.
+     */
+    public static void main(final String[] args) {
+        int numberOfMoles = 10;
+        int numAttemps = 50;
+        int gridLenght = 10;
+
+        //create a 10 by 10 grid
+        WhackAMole whackAMole = new WhackAMole(numAttemps, gridLenght);
+
+        //Place randomly a total of 10 moles
+        for (int i = 0; i < numberOfMoles; i++) {
+            Random rand = new Random();
+            int x = rand.nextInt(numberOfMoles);
+            int y = rand.nextInt(numberOfMoles);
+            if (!whackAMole.place(x, y)) {
+                System.out.println("There is already a mole at " + x + "," + y + ". Trying again!");
+                i--;
+            } else {
+                System.out.println("Mole places at " + x + "," + y);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)